### PR TITLE
Fixng ThreadAllocatedMemorySupported/Enabled handing

### PR DESCRIPTION
**Pull request by gpsre**
Fixes https://github.com/javamelody/javamelody/issues/1165

### CODE REVIEW

File: javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import javax.management.MBeanServer;
 import javax.management.MalformedObjectNameException;
 import javax.management.ObjectName;
+import javax.management.Attribute;
```
Comment (evernat) [inline javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java:30]: move up please

File: javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java [hunk 2]
```diff
@@ -40,8 +41,6 @@ final class MBeansAccessor {
 	private static final Set<String> OPERATING_SYSTEM_ATTRIBUTES = getAttributesNames(
 			OPERATING_SYSTEM);
 	private static final ObjectName THREADING = createObjectName("java.lang:type=Threading");
-	private static final boolean MBEAN_ALLOCATED_BYTES_ENABLED = getAttributesNames(THREADING)
-			.contains("ThreadAllocatedMemoryEnabled");
```
Comment (evernat) [inline javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java:44]: private static final boolean MBEAN_ALLOCATED_BYTES_ENABLED = isMbeanAllocatedBytesEnabled();

File: javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java [hunk 3]
```diff
@@ -168,4 +167,15 @@ private static Set<String> getAttributesNames(ObjectName name) {
...
 		}
 	}
+    
+    private static boolean isMbeanAllocatedBytesEnabled() {
+        try {
+            Attribute supported = (Attribute) MBEAN_SERVER.getAttribute(THREADING, "ThreadAllocatedMemorySupported");
+            Attribute enabled = (Attribute) MBEAN_SERVER.getAttribute(THREADING, "ThreadAllocatedMemoryEnabled");
+            return ((boolean) supported.getValue()) && ((boolean) enabled.getValue());
+        } catch (Exception ex) {
+            // ignored
```
Comment (evernat) [inline javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java:177]: ignored why ?

File: javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java [hunk 4]
```diff
@@ -168,4 +167,15 @@ private static Set<String> getAttributesNames(ObjectName name) {
...
+        }
+        return false;
```
Comment (evernat) [inline javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java:179]:
```suggestion
        if (getAttributesNames(THREADING).contains("ThreadAllocatedMemoryEnabled")) {
            try {
                final Attribute supported = (Attribute) MBEAN_SERVER.getAttribute(THREADING, "ThreadAllocatedMemorySupported");
                final Attribute enabled = (Attribute) MBEAN_SERVER.getAttribute(THREADING, "ThreadAllocatedMemoryEnabled");
                return ((boolean) supported.getValue()) && ((boolean) enabled.getValue());
            } catch (final JMException ex) {
                throw new IllegalStateException(ex);
            }
        }
        return false;
```

File: javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java [hunk 5]
```diff
@@ -120,7 +119,7 @@ static double getDoubleFromOperatingSystem(String attribute) {
 	}
 
 	static long getThreadAllocatedBytes(long threadId) {
-		if (!MBEAN_ALLOCATED_BYTES_ENABLED) {
+		if (!isMbeanAllocatedBytesEnabled()) {
```
Comment (evernat) [inline javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java:122]: This method may be called a lot. This change may cause a hotspot and a regression of performance, so we should probably reverse this line and change the constant instead.

Comment (evernat) [reply javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java:122]:
```suggestion
		if (!MBEAN_ALLOCATED_BYTES_ENABLED) {
```

File: javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java (hunk 3 above)
Comment (gpsre) [reply javamelody-core/src/main/java/net/bull/javamelody/internal/model/MBeansAccessor.java]: No particular thing to do with this exception. But I accepted your suggestions anyway. Thanks a lot!

**Comment by gpsre** [comment]
The PR is ready.

**Comment by evernat** [comment]
You did not test your code ??? I tested it. And there is a fatal issue which I had not imagined in review: mBeanServer#getAttribute returns a value and not a javax.management.Attribute instance.

Fixed by ba9a900

**Pull request merged**